spec/records_spec.rb

```diff
@@ -0,0 +1,254 @@
+      rec = HathifileHistory::Record.new TEST_RECID
+      rec.see TEST_HTID, TEST_YYYYMM
+      recs = records.add_record(rec)
+      expect(recs.newest_load).to eq TEST_YYYYMM
+    end
+
+    it "maintains newest_load if newer" do
+      rec = HathifileHistory::Record.new TEST_RECID
+      rec.see TEST_HTID, TEST_YYYYMM
+      recs = records.add_record(rec)
+      rec = HathifileHistory::Record.new TEST_RECID_1
+      rec.see TEST_HTID_1, TEST_EARLIER_YYYYMM
+      recs.add_record(rec)
+      expect(recs.newest_load).to eq TEST_YYYYMM
+    end
+  end
+
+  describe "#ids_from_line" do
+    context "with a well-formed line" do
+      it "parses recid and htid" do
+        htid, cid = records.ids_from_line VALID_LINE
+        expect([htid, cid]).to eq [TEST_HTID, 0]
+      end
+    end
+
+    context "with a line having Unicode issues" do
+      it "parses recid and htid" do
+        expect { records.ids_from_line INVALID_BYTE_SEQUENCE_LINE }.to raise_error ArgumentError
+      end
+    end
+
+    context "with a bogus line" do
+      it "raises" do
+        expect { records.ids_from_line UNPARSEABLE_LINE }.to raise_error NoMethodError
+      end
+    end
+  end
+
+  describe ".load_from_ndj" do
+    it "does not raise" do
+      fixture = File.join(FIXTURES_DIR, TEST_NDJ_FILE)
+      logger = NullLogger.new($stdout)
+      expect { described_class.load_from_ndj(fixture, logger: logger) }.not_to raise_error
+    end
+  end
+
+  describe "#dump_to_ndj" do
+    it "creates readable dump file" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      recs.dump_to_ndj TEST_NDJ_FILE
+      expect(File.readable?(TEST_NDJ_FILE)).to eq true
+    end
+  end
+
+  describe "#current_record_for" do
+    it "is initially empty" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      expect(recs.current_record_for(TEST_HTID)).to be_nil
+    end
+
+    it "is is populated by #compute_current_sets!" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      recs.logger = nil
+      recs.compute_current_sets! TEST_YYYYMM
+      expect(recs.current_record_for(TEST_HTID)).not_to be_nil
+    end
+  end
+
+  describe "#redirects" do
+    it "returns non-empty Hash of redirects" do
+      recs = records
+      recs.add_monthly File.join(FIXTURES_DIR, TEST_OLDER_SAMPLE_HATHIFILE_NAME)
+      recs.add_monthly File.join(FIXTURES_DIR, TEST_SAMPLE_HATHIFILE_NAME)
+      recs.compute_current_sets!(recs.newest_load)
+      recs.remove_dead_htids!
+      redirects = recs.redirects
+      expect(redirects).not_to be_empty
+      expect(redirects.keys.all? { |x| x.instance_of?(Integer) }).to eq true
+      expect(redirects.values.all? { |x| x.instance_of?(Integer) }).to eq true
+      # puts "REDIRECTS: #{recs.redirects.inspect}"
+    end
+  end
+
+  describe "#remove_dead_htids!" do
+    context "without previous call to #compute_current_sets!" do
+      it "raises" do
```

Comment:
Would it not be preferable to simply call `compute_current_sets!` if it hasn't been called, and skip the error-raising?